frontend/src/metabase/hooks/use-sequenced-content-close-handler.ts

```diff
@@ -54,14 +54,41 @@ export function shouldClosePopover(
 
 export default function useSequencedContentCloseHandler() {
   const popoverDataRef = useRef<PopoverData>();
+  // A cheap hack to prevent popovers from being closed more than 1 time
+  // in a single sequence e.g. After users click there will be 2 events; mousedown -> click.
+  const hasPopoverClosedDuringMouseSequenceRef = useRef(false);
```

Comment:
This is a new boolean flag used to tell each handler how to react
- Whether or not we should close the popover (I'm not sure if the logic should go inside `shouldClosePopover` but that's currently used by other components, so I'm not risking it.
- Whether or not to prevent `click` event from propagating to the trigger element. Imagine we're holding the click button, the popover will be closed and when we release we want to know that we just close the popover, so we shouldn't open the popover again. So while holding down the click button `hasPopoverClosedDuringMouseSequenceRef.current` will be set to `true` until the click is released.